autolens/plot/plotters/subhalo_plotters.py

```diff
@@ -0,0 +1,266 @@
+        before the subhalo added to the model (top row) and the subhalo fit which gives the largest increase in
+        Bayesian evidence on the subhalo detection grid search.
+
+        Parameters
+        ----------
+        fit_imaging_before : FitImaging
+            The fit of a `Tracer` not including a subhalo in the model to a `MaskedImaging` dataset (e.g. the
+            model-image, residual_map, chi_squared_map).
+        fit_imaging_detect : FitImaging
+            The fit of a `Tracer` with the subhalo detection grid's highest evidence model including a subhalo to a
+            `MaskedImaging` dataset (e.g. the  model-image, residual_map, chi_squared_map).
+        include : Include
+            Customizes what appears on the plots (e.g. critical curves, profile centres, origin, etc.).
+        mat_plot_2d : Plotter
+            Object for plotting PyAutoLens data-stuctures as subplots via Matplotlib.
+        """
+        number_subplots = 6
+
+        self.open_subplot_figure(number_subplots=number_subplots)
+
+        self.setup_subplot(number_subplots=number_subplots, subplot_index=1)
+
+        self.mat_plot_2d.title.kwargs["label"] = "Normalized Residuals (No Subhalo)"
+
+        fit_imaging_plotter_before = self.fit_imaging_plotter_from(
+            fit_imaging=fit_imaging_before
+        )
+
+        fit_imaging_plotter_before.figure_normalized_residual_map()
+
+        self.setup_subplot(number_subplots=number_subplots, subplot_index=2)
+
+        self.mat_plot_2d.title.kwargs["label"] = "Chi-Squared Map (No Subhalo)"
+
+        fit_imaging_plotter_before.figure_chi_squared_map()
+
+        self.setup_subplot(number_subplots=number_subplots, subplot_index=3)
+
+        self.mat_plot_2d.title.kwargs["label"] = "Source Reconstruction (No Subhalo)"
+
+        fit_imaging_plotter_before.figure_plane_image_of_plane(plane_index=1)
+
+        fit_imaging_plotter_detect = self.fit_imaging_plotter_from(
+            fit_imaging=fit_imaging_detect
+        )
+
+        self.setup_subplot(number_subplots=number_subplots, subplot_index=4)
+
+        self.mat_plot_2d.title.kwargs["label"] = "Normailzed Residuals (With Subhalo)"
+
+        fit_imaging_plotter_detect.figure_normalized_residual_map()
+
+        self.mat_plot_2d.title.kwargs["label"] = "Chi-Squared Map (With Subhalo)"
+
+        self.setup_subplot(number_subplots=number_subplots, subplot_index=5)
+
+        fit_imaging_plotter_detect.figure_chi_squared_map()
+
+        self.mat_plot_2d.title.kwargs["label"] = "Source Reconstruction (With Subhalo)"
+
+        self.setup_subplot(number_subplots=number_subplots, subplot_index=6)
+
+        fit_imaging_plotter_detect.figure_plane_image_of_plane(plane_index=1)
+
+        self.mat_plot_2d.output.subplot_to_figure()
+
+        self.mat_plot_2d.figure.close()
+
+
+def agg_max_log_likelihood_from_aggregator(aggregator):
+
+    samples = list(filter(None, aggregator.values("samples")))

```

Comment:
Samples is only used once where it is iterated through, so it's fine to leave it as a generator returned by filter rather than casting it to a list